Sources/MapboxMaps/Style/Style+Localization.swift

```diff
@@ -31,41 +31,70 @@ extension Style {
         }
     }
 
-    /// Filters through source to determine supported locale styles.
-    /// This is needed for v7 support
+    /// Returns the BCP 47 language tag supported by Mapbox Streets source v8 that is most preferred according to the given preferences.
+    /// Docs for language, region, and script codes: https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/LanguageandLocaleIDs/LanguageandLocaleIDs.html
+    internal func preferredMapboxStreetsLocalization(among preferences: [String]) -> String? {
+        let supportedLocaleIdentifiersv8 = ["ar", "en", "es", "fr", "de", "it", "pt", "ru", "zh-Hans", "zh-Hant", "ja", "ko", "vi"].map(Locale.init(identifier:))
+
+        let preferredLocales = preferences.map(Locale.init(identifier:))
+
+        let acceptsEnglish = preferredLocales.contains { $0.languageCode == "en" }
+        var availableLocales = supportedLocaleIdentifiersv8
+        if !acceptsEnglish {
```

Comment:
I took this bit directly from the navigation SDK implementation [here](https://github.com/mapbox/mapbox-navigation-ios/blob/28d13ff313cb44c47f69c6e386ec6f3dd855e63e/Sources/MapboxNavigation/VectorSource.swift#L27). However, I think it only makes sense because they fall back to the device Locale [here](https://github.com/mapbox/mapbox-navigation-ios/blob/28d13ff313cb44c47f69c6e386ec6f3dd855e63e/Sources/MapboxNavigation/VectorSource.swift#L62), which would have information about English preference. I can remove it on the Maps side. 

cc: @1ec5

Reply:
The navigation SDK code is more or less a straight Swift port of [this map SDK code](https://github.com/mapbox/mapbox-gl-native-ios/blob/ddf8e35970d1a8a46171bd798bcfa8f22d2760e0/platform/darwin/src/MGLVectorTileSource.mm#L120-L121). If I remember correctly, the rationale for this English check is that many styles, including the core Mapbox styles, hard-code English by default instead of the local language.